Group-5-Project-3/Backend
# Feature/user service test

### Summary:
This PR adds unit tests for the `UserService` class to enhance test coverage and validate core functionality.

### Changes Made:
1. Added unit tests for:
   - `loadUserByUsername()`
   - `findUserByUsername()`
   - `saveUser()`
   - `deleteUserById()`
   - `assignRole()`
   - `removeRole()`
2. Mocked dependencies using Mockito.
3. Verified core logic and edge cases.

### How to Test:
- Run `mvn test` to ensure all tests pass.
- Review the updated file at `src/test/java/com/project3/project3/service/UserServiceTest.java`.
